src/common/meta/src/peer.rs

```diff
@@ -24,6 +24,9 @@ use crate::{DatanodeId, FlownodeId};
 pub trait PeerLookupService {
     async fn datanode(&self, id: DatanodeId) -> Result<Option<Peer>, Error>;
     async fn flownode(&self, id: FlownodeId) -> Result<Option<Peer>, Error>;
+
+    /// Get all available frontends from the memory backend.
+    async fn frontends(&self) -> Result<Vec<Peer>, Error>;
 }
```

Comment:
Let's add some comments as well.
```suggestion
/// A service that can query peer given a node id.
#[async_trait::async_trait]
pub trait PeerLookupService {
    /// Returns the peer of the datanode if the metasrv receives a heartbeat from the datanode.
    ///
    /// Note: It may return an inactive peer.
    async fn datanode(&self, id: DatanodeId) -> Result<Option<Peer>, Error>;

    /// Returns the peer of the flownode if the metasrv receives a heartbeat from the flownode.
    ///
    /// Note: It may return an inactive peer.
    async fn flownode(&self, id: FlownodeId) -> Result<Option<Peer>, Error>;

    /// Get all available frontends from the memory backend.
    async fn frontends(&self) -> Result<Vec<Peer>, Error>;
}
```